Source/WebCore/Modules/permissions/PermissionStatus.h

```diff
@@ -28,40 +28,49 @@
 #include "ActiveDOMObject.h"
 #include "ClientOrigin.h"
 #include "EventTarget.h"
+#include "MainThreadPermissionObserverIdentifier.h"
+#include "Page.h"
 #include "PermissionDescriptor.h"
 #include "PermissionName.h"
 #include "PermissionObserver.h"
+#include "PermissionQuerySource.h"
 #include "PermissionState.h"
+#include "ScriptExecutionContextIdentifier.h"
+#include <atomic>
+#include <optional>
+#include <wtf/ThreadSafeRefCounted.h>
 
 namespace WebCore {
 
 class ScriptExecutionContext;
 
-class PermissionStatus final : public PermissionObserver, public ActiveDOMObject, public RefCounted<PermissionStatus>, public EventTarget  {
+class PermissionStatus final : public PermissionObserver, public ActiveDOMObject, public ThreadSafeRefCounted<PermissionStatus>, public EventTarget  {
```

Comment:
What is this change for?

Reply:
I had this because I was calling ref / deref on a PermissionStatus object on different threads. But as you suggested in your earlier comment, I can use weakThis, and that should remove the need for PermissionStatus to be ThreadSafeRefCounted.